apps/vaporgui/VSlider.cpp

```diff
@@ -0,0 +1,68 @@
+#include <iostream>
+#include <cmath>
+
+#include <vapor/VAssert.h>
+#include "VSlider.h"
+
+#define NUM_STEPS 100
+
+VSlider::VSlider( double min, double max )
+: VContainer()
+{
+    _slider = new QSlider;
+    _slider->setOrientation( Qt::Horizontal );
+    _slider->setMinimum( 0 );
+    _slider->setMaximum( NUM_STEPS );
+    SetRange( min, max );
+    SetValue( (max-min) / 2 );
+    layout()->addWidget(_slider);
+
+    connect( _slider, &QSlider::sliderMoved,
+        this, &VSlider::_sliderChangedIntermediate );
+
+    connect( _slider, &QSlider::sliderReleased,
+        this, &VSlider::_sliderChanged );
+}
+
+void VSlider::SetValue( double value ) {
+    if (value > _maxValid) value = _maxValid;
+    if (value < _minValid) value = _minValid;
+
+    value = (value-_minValid)/_stepSize;
+    _slider->blockSignals(true);
+    _slider->setValue( value );
+    _slider->blockSignals(false);
+}
+
+void VSlider::SetRange( double min, double max ) {
+    VAssert( min <= max );
+
+    _stepSize = ( max - min ) / NUM_STEPS;
```

Comment:
The `VAssert( min <= max );` line explicit says min could be equal to max, meaning `_stepSize` could be zero. Does the rest of the class handle the `_stepSize = 0.0" case properly?

Reply:
In the new code, why is step size equal to 1 if min and max are the same? 
What's the behavior of VSlider in the case when min == max??

Reply:
Instead of setting it to 1, I am now must skipping the SetValue() functionality if the _stepSize is equal to 0.  This avoids the div0 problem, and avoids an arbitrary value being applied to _stepSize in the case of 0.